File: packages/style-analyzer/fixtures/documentation-sample.ts
// @ts-nocheck
/**
 * Fetches a user profile from the API.
 *
 * @param userId - The user's unique identifier
 * @returns The user profile object
 * @throws {NotFoundError} If the user does not exist
 */
export function fetchUserProfile(userId: string): Promise<UserProfile> {
  return api.get(`/users/${userId}`);
}

/** Validates email format. */
export function validateEmail(email: string): boolean {
  return /^[^@]+@[^@]+\.[^@]+$/.test(email);
}

// This is a helper that normalizes whitespace
function normalizeWhitespace(input: string): string {
  return input.replace(/\s+/g, " ").trim();
}

export function undocumentedPublicFunction(data: unknown): void {
  console.log(data);
}

function undocumentedPrivateFunction(): number {
  return 42;
}

export class UserService {
  /**
   * Creates a new user in the database.
   *
   * @param name - Display name
   * @param email - Email address
   */
  async createUser(name: string, email: string): Promise<User> {
    return this.db.insert({ name, email });
  }

  // Quick lookup by ID
  async getUser(id: string): Promise<User | null> {
    return this.db.findById(id); // inline trailing comment
  }

  private async hashPassword(password: string): Promise<string> {
    return bcrypt.hash(password, 10);
  }
}

// Section: Utility functions
// These helpers are used across the codebase

function helperA() {
  return 1;
}

function helperB() {
  return 2; // trailing
}

File: packages/style-analyzer/fixtures/documentation-sample.py
def fetch_user_profile(user_id: str) -> dict:
    """Fetch a user profile from the API.

    Args:
        user_id: The user's unique identifier.

    Returns:
        The user profile dictionary.

    Raises:
        NotFoundError: If the user does not exist.
    """
    return api.get(f"/users/{user_id}")


def validate_email(email: str) -> bool:
    """Validate email format."""
    import re
    return bool(re.match(r"^[^@]+@[^@]+\.[^@]+$", email))


# This normalizes whitespace
def normalize_whitespace(input_str: str) -> str:
    return " ".join(input_str.split())


def undocumented_function(data):
    print(data)


class UserService:
    """Service for managing users."""

    def create_user(self, name: str, email: str) -> dict:
        """Create a new user in the database.

        Args:
            name: Display name.
            email: Email address.
        """
        return self.db.insert({"name": name, "email": email})

    def get_user(self, user_id: str) -> dict:
        # Quick lookup by ID
        return self.db.find_by_id(user_id)
